src/components/NavigationBar/type.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { FunctionComponent, useState } from "react";
+import { Icon } from "react-feather";
+import { NavLink } from "react-router-dom";
+
+export interface NavigationBarProps {
+  navigationItems: NavigationItem[];
+}
+
+export enum NavigationItemType {
+  NavigationLink = "NavigationLink",
+  DropDownList = "DropDownList",
+  LabelButton = "LabelButton",
+  IconButton = "IconButton",
+}
+
+interface NavigationLink {
+  schema: NavigationItemType.NavigationLink;
+  id: string;
+  label: string;
+  className?: string;
+  path: string;
+}
+
+interface DropDownList extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.DropDownList;
+  dropdownItems: { id: string; label: string; path: string }[];
+}
+
+interface LabelButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.LabelButton;
+  onClick?: () => void;
+}
+
+interface IconButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.IconButton;
+  icon: Icon;
+}
+
+export type NavigationItem = NavigationLink | DropDownList | LabelButton | IconButton;
+
+export const NavigationBarItem: FunctionComponent<{ item: NavigationItem; onClick: (isOn: boolean) => void }> = ({
+  item,
+  onClick,
+}) => {
+  switch (item.schema) {
+    case NavigationItemType.IconButton:
+      return <IconButton item={item} onClick={onClick} />;
+    case NavigationItemType.LabelButton:
+      return <LabelButton item={item} onClick={onClick} />;
+    case NavigationItemType.DropDownList:
+      return <DropDownList item={item} onClick={onClick} />;
+    default:
+      return <NavigationLink item={item} onClick={onClick} />;
+  }
+};
+
+const NavigationLink: FunctionComponent<{ item: NavigationLink; onClick: (isOn: boolean) => void }> = ({
+  item,
+  onClick,
```

Comment:
these are not types, shift them to a NavigationBar or create a new folder?

Reply:
Shifted the react component to NavigationBarItem